### PR TITLE
cloud: convert BuildRecords into update proto objects

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism,

Please review the following commits I made in branch nicks/update:

350c98c1b831bf8c8b6f2c1f92f73c17e9c6c021 (2019-10-14 21:27:20 -0400)
cloud: convert BuildRecords into update proto objects

### CODE REVIEW

File: internal/cloud/update_uploader.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	// Do a quick check to see if any builds have completed since we last checked.
+	if state.CompletedBuildCount == 0 || state.CompletedBuildCount <= u.lastCompletedBuildCount {
+		return nil
+	}
+
+	// OK, we know we have work to do!
+
+	highWaterMark := u.lastFinishTime
```
Comment (landism) [inline internal/cloud/update_uploader.go]: is there anything protecting this function from getting invoked in parallel by multiple calls to `OnChange` and processing the same set of builds?

similarly, do we need synchronization on accessing `u`'s fields?

File: internal/cloud/update_uploader.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+// Check the engine state to see if we have any updates.
+func (u *UpdateUploader) makeUpdates(st store.RStore) []update {
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	// Check if this feature is disabled
+	if !state.Features[feature.UpdateHistory] {
+		return nil
+	}
+
```
Comment (landism) [inline internal/cloud/update_uploader.go]: let's also return `if state.TeamName == ""`

File: internal/feature/flags.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ const (
 const MultipleContainersPerPod = "multiple_containers_per_pod"
 const Events = "events"
 const Snapshots = "snapshots"
+const UpdateHistory = "update-history"
```
Comment (landism) [inline internal/feature/flags.go]: nit: our other feature names are `_` rather than `-`

File: internal/cloud/update_uploader.go (hunk 1 above)
Comment (nicks) [reply internal/cloud/update_uploader.go]: from:
https://github.com/windmilleng/tilt/blob/master/internal/store/subscriber.go#L11

```
// Subscribers do not need to be thread-safe. The Store will only
// call OnChange for a given subscriber when the last call completes.
```

File: internal/cloud/update_uploader.go (hunk 2 above)
Comment (nicks) [reply internal/cloud/update_uploader.go]: oh, we're not supporting update-history for non-teams?

File: internal/cloud/update_uploader.go (hunk 1 above)
Comment (landism) [reply internal/cloud/update_uploader.go]: Whoops, apparently my approach to subscriber concurrency has been off. I've been spinning blocking work off to a goroutine and then dealing with locking, unaware the whole thing is already in a thread-safe goroutine.

**Pull request merged**